src/Microsoft.Graph.Core/Requests/BaseFunctionMethodRequestBuilder.cs

```diff
@@ -48,25 +44,21 @@ public abstract class BaseFunctionMethodRequestBuilder<T> : BaseRequestBuilder w
         /// <returns>The built request.</returns>
         public T Request(IEnumerable<Option> options = null)
         {
-            string fnUrl = this.RequestUrl;
-
-            if (this.passParametersInQueryString && this._queryOptions.Count > 0)
+            // Validate the query options according to 
+            // http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_InvokingaFunction
+            if (options != null)
             {
-                if (options == null)
-                {
-                    options = this._queryOptions;
-                }
-                else
+                var unexpectedQueryOptions = options.OfType<QueryOption>().Any<QueryOption>(qp => 
```

Comment:
Second type annotation is redundant.
```suggestion
                var unexpectedQueryOptions = options.OfType<QueryOption>().Any(qp => 
```

Reply:
Done